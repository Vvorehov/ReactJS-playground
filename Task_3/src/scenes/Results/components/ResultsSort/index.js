export { default as ResultsSort } from './ResultsSort';