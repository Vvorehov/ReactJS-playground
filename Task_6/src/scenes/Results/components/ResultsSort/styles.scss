.result-sort {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 2rem;
}

.result-sort_title {
  padding-right: 2rem;
  font-size: 1.4rem;
}

.result-sort_btn {
  padding: 0 1rem;
}