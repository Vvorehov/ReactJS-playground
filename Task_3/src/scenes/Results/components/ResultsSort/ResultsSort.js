import React, {Component} from 'react';
import { Radio } from './../../../Radio';
import './styles.scss'

class ResultsSort extends Component {
  render() {
    return (
      <form className="result-sort">
        <div className="result-sort_title">Sort by</div>
        <div className="result-sort_btn">
          <Radio name="result-sort" label="Release date" value="date" isChecked={true} mod="sort"/>
        </div>
        <div className="result-sort_btn">
          <Radio name="result-sort" label="Rating" value="rating" mod="sort"/>
        </div>
      </form>
    )
  }
}

export default ResultsSort;