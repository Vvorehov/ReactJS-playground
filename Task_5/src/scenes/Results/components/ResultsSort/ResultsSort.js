import React, {Component} from 'react';
import { Radio } from './../../../Radio';
import actionTypes from "../../../../actions";
import './styles.scss'

class ResultsSort extends Component {
  constructor(props) {
    super(props);
    this.handleOnChange = this.handleOnChange.bind(this);
  }

  handleOnChange(e) {
  }

  render() {
    return (
      <form className="result-sort">
        <div className="result-sort_title">Sort by</div>
        <div className="result-sort_btn">
          <Radio name="result-sort" label="Release date" onChange="" value="SORT_BY_RELEASE_DATE" isChecked={true} mod="sort"/>
        </div>
        <div className="result-sort_btn">
          <Radio name="result-sort" label="Rating" onChange="" value="SORT_BY_RATING" mod="sort"/>
        </div>
      </form>
    )
  }
}

export default ResultsSort;