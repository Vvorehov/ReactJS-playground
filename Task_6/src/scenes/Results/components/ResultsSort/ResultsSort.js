import React, {Component} from 'react';
import { Radio } from '../../../Radio';
import actionTypes from "../../../../constants/actionsTypes.js";

import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import * as FetchData from "../../../../actions/fetchData.js";
import './styles.scss'

class ResultsSort extends Component {
  constructor(props) {
    super(props);
    this.handleOnChange = this.handleOnChange.bind(this);
  }

  handleOnChange(e) {
    this.props.fetchSearchResults(this.props.searchData, e.target.value);
  }

  render() {
    return (
      <form className="result-sort">
        <div className="result-sort_title">Sort by</div>
        <div className="result-sort_btn">
          <Radio name="result-sort" label="Release date" value="release_date" onChange={this.handleOnChange} isChecked={this.props.sortData.sortBy === "DATE"} mod="sort"/>
        </div>
        <div className="result-sort_btn">
          <Radio name="result-sort" label="Rating" value="vote_average" onChange={this.handleOnChange} isChecked={this.props.sortData.sortBy === "RATING"} mod="sort"/>
        </div>
      </form>
    )
  }
}


function mapStateToProps(state) {
  return {
    sortData: state.sortData,
    searchData: state.searchData
  }
}

function mapDispatchToProps(dispatch) {
  return {
    fetchSearchResults: bindActionCreators(FetchData.fetchSearchResults, dispatch)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ResultsSort)