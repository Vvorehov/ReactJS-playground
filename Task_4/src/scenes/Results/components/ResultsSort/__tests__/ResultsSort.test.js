import React from "react";
import ResultsSort from "../ResultsSort";
import TestRenderer from 'react-test-renderer';

describe('ResultsSort layout complete', () => {
    test('Render correct', () => {

        const component = TestRenderer.create(
            <ResultsSort />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

