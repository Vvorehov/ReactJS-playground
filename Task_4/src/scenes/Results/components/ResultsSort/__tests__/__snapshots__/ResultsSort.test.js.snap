// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsSort layout complete Render correct 1`] = `
<form
  className="result-sort"
>
  <div
    className="result-sort_title"
  >
    Sort by
  </div>
  <div
    className="result-sort_btn"
  >
    <label
      className="radio radio--sort"
    >
      <input
        defaultChecked={true}
        name="result-sort"
        onChange={[Function]}
        type="radio"
        value="date"
      />
      <span>
        Release date
      </span>
    </label>
  </div>
  <div
    className="result-sort_btn"
  >
    <label
      className="radio radio--sort"
    >
      <input
        defaultChecked={undefined}
        name="result-sort"
        onChange={[Function]}
        type="radio"
        value="rating"
      />
      <span>
        Rating
      </span>
    </label>
  </div>
</form>
`;
